internal/client/prism.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	prismcommonapi "github.com/nutanix/ntnx-api-golang-clients/prism-go-client/v4/models/common/v1/config"
+	prismapi "github.com/nutanix/ntnx-api-golang-clients/prism-go-client/v4/models/prism/v4/config"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/nutanix-cloud-native/prism-go-client/utils"
+	v4 "github.com/nutanix-cloud-native/prism-go-client/v4"
+)
+
+type WaitForTaskCompletionOpts struct {
+	PollInterval  time.Duration
+	Timeout       time.Duration
+	PollImmediate bool
+}
+
+func (o WaitForTaskCompletionOpts) PollIntervalOrDefault() time.Duration {
```

Comment:
```suggestion
func (o WaitForTaskCompletionOpts) PollInterval() time.Duration {
```

Why to have the `OrDefault` suffix? A function caller have no way (or need) to differentiate between provided and default.

Reply:
Fair, good ol' Copilot helping out 😂

Reply:
Couldn't just rename due to conflicts with the field names. Instead in [`79c8643` (#13)](https://github.com/nutanix-cloud-native/cluster-api-ipam-provider-nutanix/pull/13/commits/79c86439e35658e435a021ddb8cd19dd9657ac95) I removed the funcs, referenced the fields directly, and provided defaults as consts.